modules/eks-managed-node-group/main.tf

```diff
@@ -30,7 +30,7 @@ module "user_data" {
 ################################################################################
 
 locals {
-  use_custom_launch_template = var.launch_template_name != ""
+  use_custom_launch_template = var.create_launch_template || var.launch_template_name != ""
```

Comment:
this is the fix for ensuring the custom launch template is used when the sub-module is used externally

there are 4 scenarios here which is why it can be a bit weird to understand:
1. `var.create_launch_template = false && var.launch_template_name == ""` => EKS MNG will use its own default LT
2. `var.create_launch_template = false && var.launch_template_name == "something"` => User provided custom LT will be used
3. `var.create_launch_template = true && var.launch_template_name == ""` => Custom LT will be used, module will provide a default name
4. `var.create_launch_template = true && var.launch_template_name == "something"` => Custom LT will be used, LT name is provided by user

Reply:
Maybe put your comment right above this line as a description for this "light magic"? It will be helpful to know the reason in some time.